app/models/hackathon/digest/listings.rb

```diff
@@ -7,24 +7,32 @@ module Hackathon::Digest::Listings
     has_many :listed_subscriptions, through: :listings, source: :subscription
 
     before_validation :build_candidate_listings, on: :create, if: -> { listings.empty? }
+
     validates_length_of :listings, minimum: 1, on: :create, message: "must not be empty"
+    validate :minimum_one_unseen_listing, on: :create, if: -> { errors.none? }
   end
 
   private
 
   LISTING_CRITERIA = [Criterion::Location]
   MAX_LISTINGS = 8
 
+  def build_candidate_listings
+    applicable_listings.each do |result|
+      listings.build hackathon: result[:hackathon], subscription: result[:subscription]
+    end
+  end
```

Comment:
@northeastprince I'm thinking about how we can do the filtering in the mailer, what would be the best way to add a field to a listing that is like "new = true / false"? that'd make filtering super easy in the view and we can use the logic in `minimum_one_unseen_listing` to conditionally apply set the tag?

Reply:
How's this?

```ruby
# Hackathon::Digest::Listing

def previously_listed_for_recipient?(recipient: digest.recipient) do
  exists?(hackathon: hackathon, recipient: recipient)
end
```

Reply:
I was more thinking of something like a database field, would that make sense here? I want for when we build listings us to be able to differentiate between new ones and old / stale ones

Reply:
alright, pushed!